extensions/wikia/ArticleVideo/bower_components/html5-skin/js/components/utils.js

```diff
@@ -404,12 +421,132 @@ var Utils = {
    * See https://facebook.github.io/react/tips/dangerously-set-inner-html.html
    *
    * @function createMarkup
+   * @param {String} html - html to be sanitized
    * @returns {Object} Wrapper object for sanitized markup.
    */
   createMarkup: function(html) {
     return {__html: html};
   },
 
+  /**
+   * Deep merge arrays and array values
+   *
+   * @function arrayDeepMerge
+   * @param {Array} target - An array that will receive new items if additional items are passed
+   * @param {Array} source - An array containing additional items to merge into target
+   * @param {Object} optionsArgument - parameters passed to parent DeepMerge function, i.e. -
+   *        arrayMerge - https://github.com/KyleAMathews/deepmerge#arraymerge
+   *        clone - https://github.com/KyleAMathews/deepmerge#clone
+   *        arrayUnionBy - key used to compare Objects being merged, i.e. button name
+   *        arrayFusion - method used to merge arrays ['replace', 'deepmerge']
+   *        buttonArrayFusion - method used to merge button array ['replace', 'prepend', 'deepmerge']
+   *        arraySwap - swaps target/source
+   * @returns {Array} new merged array with items from both target and source
+   */
+  arrayDeepMerge: function(target, source, optionsArgument) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Function has a complexity of 16 which is greater than 10 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AFunctionComplexity)